src/NodeAnalyzer/QueryBuilderAnalyzer.php

```diff
@@ -26,6 +31,19 @@ protected static function queryBuilderType(): ObjectType
         return new ObjectType('Illuminate\Contracts\Database\Query\Builder');
     }
 
+    protected static function eloquentQueryBuilderType(): ObjectType
+    {
+        return new ObjectType('Illuminate\Contracts\Database\Eloquent\Builder');
```

Comment:
This should use the actual builder, not the contract (I hate the contracts: they are empty and not true contracts, and the Relation also implements the contract which is not great)


```suggestion
        return new ObjectType('Illuminate\Database\Eloquent\Builder');
```

Reply:
Fair. My slight worry is that most will type hint the contract still and it might then cause the rule to miss things. In the analyser it should be fine.